packages/twenty-server/src/engine/core-modules/billing/stripe/services/stripe-subscription-schedule.service.ts

```diff
@@ -0,0 +1,59 @@
+/* @license Enterprise */
+
+import { Injectable, Logger } from '@nestjs/common';
+
+import type Stripe from 'stripe';
+
+import { StripeSDKService } from 'src/engine/core-modules/billing/stripe/stripe-sdk/services/stripe-sdk.service';
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+import { SubscriptionWithSchedule } from 'src/engine/core-modules/billing/types/billing-subscription-with-schedule.type';
+
+@Injectable()
+export class StripeSubscriptionScheduleService {
+  protected readonly logger = new Logger(
+    StripeSubscriptionScheduleService.name,
+  );
+  private readonly stripe: Stripe;
+
+  constructor(
+    private readonly twentyConfigService: TwentyConfigService,
+    private readonly stripeSDKService: StripeSDKService,
+  ) {
+    if (!this.twentyConfigService.get('IS_BILLING_ENABLED')) {
+      return;
+    }
+    this.stripe = this.stripeSDKService.getStripe(
+      this.twentyConfigService.get('BILLING_STRIPE_API_KEY'),
+    );
+  }
+
+  async findOrCreateSubscriptionSchedule(
+    subscription: SubscriptionWithSchedule,
+  ) {
+    return (
+      subscription.schedule ??
+      (await this.createScheduleFromSubscription(subscription.id))
```

Comment:
this is hard to debug when we use break point

if (isDefined(sub.schedule) { return sub.schedule }

const createdSchedule = await...

return createdScheudl